src/main/java/org/benetech/servicenet/repository/ActivityRepository.java

```diff
@@ -216,11 +248,20 @@ private Predicate addDateFilter(FiltersActivityDTO filtersActivityDTO, Predicate
         Join<ActivityInfo, OrganizationMatch> matchJoin = root.join(ORGANIZATION_MATCHES, JoinType.LEFT);
         predicate = cb.and(predicate, cb.equal(matchJoin.get(HIDDEN), filtersActivityDTO.getHiddenFilter()));
 
-        if (CollectionUtils.isNotEmpty(filtersActivityDTO.getPartnerFilterList())) {
+        boolean hasPartnerFilters = (CollectionUtils.isNotEmpty(filtersActivityDTO.getPartnerFilterList()));
+        if (hasPartnerFilters || partnerRecords) {
             Join<OrganizationMatch, Organization> matchOrgJoin = matchJoin.join(PARTNER_VERSION, JoinType.LEFT);
-            Join<Organization, SystemAccount> accountJoin = matchOrgJoin.join(ACCOUNT, JoinType.LEFT);
+            Join<Organization, SystemAccount> matchAccountJoin = matchOrgJoin.join(ACCOUNT, JoinType.LEFT);
 
-            predicate = cb.and(predicate, accountJoin.get(ID).in(filtersActivityDTO.getPartnerFilterList()));
+            if (partnerRecords) {
+                predicate = cb.and(predicate, cb.notEqual(matchAccountJoin.get(ID), ownerId));
```

Comment:
if there are any partner filters I think it should be enough to make sure the ownerId is not in the filtersActivityDTO.getPartnerFilterList and if there are none you can add this predicate

Reply:
I don't think it would work, in that case partnerFilterList filters on matches, not the selected organization

Reply:
OK, I see now